go/types.go

```diff
@@ -196,22 +239,58 @@ type LogEntry struct {
 	Chain    []ASN1Cert
 }
 
+// SHA256Hash represents the output from the SHA256 hash function.
+type SHA256Hash [sha256.Size]byte
+
+// FromBase64String populates the SHA256 struct with the contents of the base64 data passed in.
+func (s *SHA256Hash) FromBase64String(b64 string) error {
+	bs, err := base64.StdEncoding.DecodeString(b64)
+	if err != nil {
+		return fmt.Errorf("failed to unbase64 LogID: %v", err)
+	}
+	if len(bs) != sha256.Size {
+		return fmt.Errorf("invalid SHA256 length, expected 32 but got %d", len(bs))
+	}
+	copy(s[:], bs)
+	return nil
+}
+
+// Base64String returns the base64 representation of this SHA256Hash.
+func (s SHA256Hash) Base64String() string {
+	return base64.StdEncoding.EncodeToString(s[:])
+}
+
+// MarshalJSON implements the json.Marshaller interface for SHA256Hash.
+func (s SHA256Hash) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + s.Base64String() + `"`), nil
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface.
+func (s *SHA256Hash) UnmarshalJSON(b []byte) error {
+	if b[0] != '"' || b[len(b)-1] != '"' {
+		return fmt.Errorf("expected a quoted string, but got '%s'", string(b))
+	}
+	content := string(b[1 : len(b)-1])
```

Comment:
This seems a bit iffy, with regard to JSON (quoting and other wackiness)... Would it be possible to leverage `UnmarshalJSON` for `string`, instead? So it would just be something like `content.UnmarshalJSON(b)`? Presumably, there's handling in there for the cases where it's not a string, and so on...